report/4.Solution/ParserImplementation.tex

```diff
@@ -115,7 +118,83 @@ \subsection{Implementing a recursive decent parser}
+\end{lstlisting}
+
+As seen in snippet \ref{lis:parsePotCall}, the method first determines if the current token is a lambda expression, signified by a \textit{blockStart-token} (3).
+Should it be a lambda expression, the parser determines whether or not there is a functions call within the lambda that needs to be parsed as well (7).
+If there is not a function-call within the lambda, the lambda expression itself is returned from the method (10). 
+Should the token be an identifier, the parser determines if the identifier signifies a function, in which case it is determined that a function-call is being made (16).
+The method looks for the needed arguments for the function, and parse them as expressions.
+The method then returns the function-call with its corresponding arguments (18).
+If the identifier does not signify a function, it is then parsed as a regular expression.
+
+All expressions that are determined to not be function-calls, are parsed by the method \textbf{\textit{parseExpressionAtomic}} seen in snippet \ref{lis:parseExpr}.
+
+\begin{lstlisting}[language=Kotlin,label=lis:parseExpr,caption=A simplified version of the parseExpressionAtomic method from the parser.]
+private fun parseExpressionAtomic(): AstExpression =
+    when (current.token) {
+        Token.SpecialChar.SquareBracketStart -> parseListDeclaration()
+        Token.SpecialChar.ParenthesesStart -> {
+            if (isLambdaParameters()) {
+                parseLambdaDeclaration()
+            } else {
+                if(upcomingTuple()){
+                    parseTupleExpression()
+                } else {
+                    parseExpression()
+                }
+            }
+        }
+        Token.Literal -> acceptLiteral()
+        Token.SpecialChar.Colon -> {
+        	accept<Token.SpecialChar.Colon>()
+        	val token = accept<Token.Identifier>()
+        	retrieveSymbol(token.value).handle(..handle token as function..)
+        }
+        Token.Identifier -> {
+            val token = accept<Token.Identifier>()
+            retrieveSymbol(token.value).handle(..Handle identifier..)
+        }
+        else -> unexpectedTokenError(current.token)
+    }
+            
+\end{lstlisting}
 
+The method itself deals with determining the type of the expression that is to be parsed, and then parsing it atomically.
+All expressions that begin with the '\texttt{[}' character are determined to be list declarations and are parsed as such (3).\\
+Expressions beginning with a parentheses can either be a lambda declaration (6), a tuple (9) or simply an expression encapsulated in parenthesis (11).\\
+Literals are simply accepted (15).\\
+Expressions beginning with a colon is can only be functions passed as arguments, and are handled accordingly (16).\\
+Identifiers can either signify values or function-calls.
+However, as potential function-calls have already been handled at this point, the identifier is determined to signify a value and can be parsed accordingly (21).
```

Comment:
If my last comment leads to changes, it should be reflected in this sentence as well.